packages/components/src/text/mixins.js

```diff
@@ -0,0 +1,149 @@
+/**
+ * External dependencies
+ */
+import css from '@emotion/css';
+/**
+ * Internal dependencies
+ */
+import { fontFamily } from './font-family';
+
+const fontWeightNormal = `font-weight: 400;`;
+const fontWeightSemibold = `font-weight: 600;`;
+
+const title = `
+  ${ fontWeightNormal }
+`;
+
+const titleLarge = `
+	font-size: 32px;
+	line-height: 40px;
+`;
+
+const titleMedium = `
+	font-size: 24px;
+	line-height: 32px;
+`;
+
+const titleSmall = `
+	font-size: 20px;
+	line-height: 28px;
+`;
+
+const subtitle = `
+	${ fontWeightSemibold }
+	font-size: 14px;
+	line-height: 20px;
+`;
+
+const subtitleLarge = `
+	font-size: 16px;
+	line-height: 24px;
+`;
+
+const subtitleSmall = `
+	font-size: 14px;
+	line-height: 20px;
+`;
+
+const body = `
+	${ fontWeightNormal }
+`;
+
+const bodyLarge = `
+	font-size: 16px;
+	line-height: 24px;
+`;
+
+const bodySmall = `
+	font-size: 14px;
+	line-height: 20px;
+`;
+
+const button = `
+  ${ fontWeightSemibold }
+  font-size: 14px;
+  line-height: 20px;
+`;
+
+const caption = `
+	${ fontWeightNormal }
+	font-size: 12px;
+	line-height: 16px;
+`;
+
+const label = `
+	${ fontWeightSemibold }
+	font-size: 12px;
+	line-height: 16px;
+`;
+
+/**
+ * @typedef {"title.large" | "title.medium" | "title.small" | "subtitle" | "subtitle.small" | "body" | "body.small" | "button" | "caption" | "label"} TextVariant
+ */
+
+/**
+	* @param {TextVariant} variantName
+	*/
+const variant = ( variantName ) => {
+	switch ( variantName ) {
+		case 'title.large':
+			return css`
+				${ title }
+				${ titleLarge }
+			`;
+		case 'title.medium':
+			return css`
+				${ title }
+				${ titleMedium }
+			`;
+		case 'title.small':
+			return css`
+				${ title }
+				${ titleSmall }
+			`;
+
+		case 'subtitle':
+			return css`
+				${ subtitle }
+				${ subtitleLarge }
+			`;
+		case 'subtitle.small':
+			return css`
+				${ subtitle }
+				${ subtitleSmall }
+			`;
+
+		case 'body':
+			return css`
+				${ body }
+				${ bodyLarge }
+			`;
+		case 'body.small':
+			return css`
+				${ body }
+				${ bodySmall }
+			`;
+
+		case 'button':
+			return button;
+
+		case 'caption':
+			return caption;
+
+		case 'label':
+			return label;
+	}
+};
+
+/**
+ * @typedef {Object} TextProps
+ * @property {TextVariant} variant
+ */
+
+/**
+ * @param {TextProps} props
```

Comment:
Nice, I'm wondering if Storybook 5.3 will be able to pull this JSDoc definition. It seems like it should be smart enough :)

Reply:
What does 5.3 do with JSDoc??!

Reply:
<img src="https://miro.medium.com/max/3958/1*XGUPJaHY7riY2qNKtEl_3Q.png" />

See more at https://medium.com/storybookjs/rich-docs-with-storybook-mdx-61bc145ae7bc.

Reply:
In 5.2 it supports very basic parsing:

![Screen Shot 2019-12-13 at 10 04 00](https://user-images.githubusercontent.com/699132/70787860-ebb11000-1d8f-11ea-8943-5cbdbfbafe78.png)

See: https://wordpress.github.io/gutenberg/?path=/docs/components-toolbargroup--default.

Reply:
Oh cool. I can't find much details on it. If not, `react-docgen-typescript-loader` would probably work to generate prop tables from the JSDoc too. https://storybook.js.org/docs/configurations/typescript-config/